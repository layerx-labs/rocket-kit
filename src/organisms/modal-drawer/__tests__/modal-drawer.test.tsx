import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ModalDrawer from '..';
import userEvent from '@testing-library/user-event';

describe('ModalDrawer', () => {
  it('renders', () => {
    const { asFragment } = render(
      <ModalDrawer
        isShowing
        title="Checkout"
        closeValue={'Close'}
        hide={() => {}}
      >
        <p>ground round corned beef</p>
      </ModalDrawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('is not showing', () => {
    render(
      <ModalDrawer
        isShowing={false}
        title="Checkout"
        closeValue={'Close'}
        hide={() => {}}
      >
        <p>ground round corned beef</p>
      </ModalDrawer>
    );

    expect(
      screen.queryByText(/ground round corned beef/)
    ).not.toBeInTheDocument();
  });

  it('is showing', () => {
    const onClick = jest.fn();
    render(
      <ModalDrawer
        isShowing={true}
        title="Checkout"
        closeValue={'Submit'}
        hide={onClick}
      >
        <p>ground round corned beef</p>
      </ModalDrawer>
    );

    screen.getByText(/ground round corned beef/i);
    screen.getByText(/checkout/i);

    userEvent.click(screen.getByRole('button', { name: /submit/i }));
    expect(onClick).toBeCalledTimes(1);
  });
});
