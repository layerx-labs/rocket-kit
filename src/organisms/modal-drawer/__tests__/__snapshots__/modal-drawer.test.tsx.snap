// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDrawer renders 1`] = `<DocumentFragment />`;
